MSAL/src/native_auth/controllers/jit/MSALNativeAuthJITController.swift

```diff
@@ -0,0 +1,300 @@
+    private let jitResponseValidator: MSALNativeAuthJITResponseValidating
+    private let signInController: MSALNativeAuthSignInControlling
+
+    // MARK: - Init
+
+    init(
+        clientId: String,
+        jitRequestProvider: MSALNativeAuthJITRequestProviding,
+        jitResponseValidator: MSALNativeAuthJITResponseValidating,
+        signInController: MSALNativeAuthSignInControlling
+    ) {
+        self.jitRequestProvider = jitRequestProvider
+        self.jitResponseValidator = jitResponseValidator
+        self.signInController = signInController
+        super.init(clientId: clientId)
+    }
+
+    convenience init(config: MSALNativeAuthConfiguration, cacheAccessor: MSALNativeAuthCacheInterface) {
+        self.init(
+            clientId: config.clientId,
+            jitRequestProvider: MSALNativeAuthJITRequestProvider(
+                requestConfigurator: MSALNativeAuthRequestConfigurator(config: config)),
+            jitResponseValidator: MSALNativeAuthJITResponseValidator(),
+            signInController: MSALNativeAuthSignInController(config: config, cacheAccessor: cacheAccessor)
+        )
+    }
+
+    func getJITAuthMethods(
+        continuationToken: String,
+        context: MSALNativeAuthRequestContext
+    ) async -> JITGetJITAuthMethodsControllerResponse {
+        let event = makeAndStartTelemetryEvent(id: .telemetryApiIdJITIntrospect, context: context)
+        let result = await performAndValidateIntrospectRequest(continuationToken: continuationToken, context: context)
+        let telemetryInfo = TelemetryInfo(event: event, context: context)
+        switch result {
+        case .authMethodsRetrieved(let newContinuationToken, let authMethods):
+            return .init(.selectionRequired(
+                authMethods: authMethods.map({$0.toPublicAuthMethod()}),
+                continuationToken: newContinuationToken
+            ), correlationId: telemetryInfo.context.correlationId(),
+            telemetryUpdate: { [weak self] result in
+                self?.stopTelemetryEvent(telemetryInfo.event, context: telemetryInfo.context, delegateDispatcherResult: result)
+            })
+        case .error(let error):
+            MSALLogger.logPII(
+                level: .error,
+                context: telemetryInfo.context,
+                format: "JIT: an error occurred after calling /introspect API: \(MSALLogMask.maskPII(error))"
+            )
+            stopTelemetryEvent(telemetryInfo, error: error)
+            return .init(.error(error: error), correlationId: context.correlationId())
+        }
+    }
+
+    func requestJITChallenge(
+        continuationToken: String,
+        authMethod: MSALAuthMethod,
+        verificationContact: String?,
+        context: MSALNativeAuthRequestContext
+    ) async -> JITRequestChallengeControllerResponse {
+        let event = makeAndStartTelemetryEvent(id: .telemetryApiIdJITChallenge, context: context)
```

Comment:
can you please set the verificationContact value here? We want to highlight this behaviour so it's clear what we do when verification contact is nil.
```suggestion
        let event = makeAndStartTelemetryEvent(id: .telemetryApiIdJITChallenge, context: context)
        // We use the authentication method loginHint value if the user has not specified a verification contact.
        let verificationContact = verificationContact ?? authMethod.loginHint
```